src/main/java/sim/coffee/Server.java

```diff
@@ -1,14 +1,19 @@
 package sim.coffee;
 
-public class Server implements Runnable {
+public class Server implements Runnable{
+    private SharedQueue queue;
 
-    public Server(SharedQueue orders) {
+    public Server(SharedQueue queue) {
+        this.queue = queue;
     }
 
-    @Override
     public void run() {
-        // TODO Auto-generated method stub
-
+        while (!queue.getDone()) {
+            try {
+                queue.getCustomerOrder();
+                Thread.sleep(100);
```

Comment:
To adapt the sleep you just get the array returned by getCusomterOrder and then use the size of it

Reply:
Can't seem to get this to work. Going to create a method on SharedQueue and see how it goes tomorrow in the meeting.